editor/editor_properties_array_dict.cpp

```diff
@@ -898,255 +972,45 @@ void EditorPropertyDictionary::update_property() {
 		int total_amount = page_index == max_page ? amount + 2 : amount; // For the "Add Key/Value Pair" box on last page.
 
 		VBoxContainer *add_vbox = nullptr;
-		double default_float_step = EDITOR_GET("interface/inspector/default_float_step");
 
 		for (int i = 0; i < total_amount; i++) {
 			String prop_name;
-			Variant key;
 			Variant value;
 
 			if (i < amount) {
 				prop_name = "indices/" + itos(i + offset);
-				key = dict.get_key_at_index(i + offset);
 				value = dict.get_value_at_index(i + offset);
 			} else if (i == amount) {
 				prop_name = "new_item_key";
 				value = object->get_new_item_key();
-			} else if (i == amount + 1) {
+				if (value.get_type() == Variant::NIL && key_type != Variant::NIL) {
+					VariantInternal::initialize(&value, key_type);
+					object->set_new_item_key(value);
+				}
+			} else {
 				prop_name = "new_item_value";
 				value = object->get_new_item_value();
+				if (value.get_type() == Variant::NIL && value_type != Variant::NIL) {
+					VariantInternal::initialize(&value, value_type);
+					object->set_new_item_value(value);
+				}
```

Comment:
why are those only present for new_key/new_value?

Reply:
The values in the dictionary itself will have already been initalized by this point. Only the new key/value fields need to be initalized explicitly

Reply:
if it is about initialisation shouldn't it be in the initialisation function you created?

Reply:
That's what I assumed at first, but it turns out that would only initalize it for a single new key/value. As soon as they're added to the dictionary, the new key/value get reset back to a plain `Variant` & require initalization once more. Making the initalization happen in this check instead will guarantee that the editor property will always have an initalized new key/value to work with

Reply:
what you talk about seems like a regression from #88636. you might have wrongly merged it. Before the linked PR it was reverting them back too variant null but after this PR it should initialize them to the right type.

Reply:
I'm describing the behavior it had when this part of the PR was initially set up, so it being changed since then is entirely reasonable. I'll have to see if that reset behavior is still in place anywhere

Reply:
Yep, this was indeed fixed! The relevant code has been moved to `setup` instead